File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        Map<Integer, Integer> seen = new HashMap<>();

        for (int i = 0; i < nums.length; i++){
            int needed = target - nums[i];

            if (seen.containsKey(needed)){
                return new int[]{i,seen.get(needed)};
            }

            seen.put(nums[i], i);
        }

        return new int[]{} ;
        
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}

        for i in range(len(nums)):
            needed = target - nums[i]

            if needed in seen:
                return [i, seen[needed]]

            seen[nums[i]] = i

        return []
        